stdlib/Equals.dora

```diff
@@ -0,0 +1,7 @@
+trait Equals {
+  fun equals(other: Self) -> Bool;
```

Comment:
I think it would be better to not have the distinction between `equals` and `==`.